Gulpfile.ts

```diff
@@ -759,23 +759,53 @@ gulp.task(nodeServerOutFile, false, [servicesFile], () => {
         .pipe(gulp.dest(path.dirname(nodeServerOutFile)));
 });
 
+import convertMap = require("convert-source-map");
+import sorcery = require("sorcery");
+declare module "convert-source-map" {
+    export function fromSource(source: string, largeSource?: boolean): SourceMapConverter;
+}
+
 gulp.task("browserify", "Runs browserify on run.js to produce a file suitable for running tests in the browser", [servicesFile], (done) => {
     const settings: tsc.Settings = getCompilerSettings({
-        outFile: "built/local/bundle.js"
+        outFile: "built/local/bundle.js",
     }, /*useBuiltCompiler*/ true);
     return gulp.src(harnessSources)
         .pipe(newer("built/local/bundle.js"))
         .pipe(sourcemaps.init())
         .pipe(tsc(settings))
         .pipe(through2.obj((file, enc, next) => {
-            browserify(intoStream(file.contents))
+            const originalMap = file.sourceMap;
+            const prebundledContent = file.contents.toString();
+            originalMap.sources = originalMap.sources.map(s => path.resolve(s)); // Make paths absolute to help sorcery deal with all the terrible paths being thrown around

```

Comment:
Scratch that, `map(path.resolve)` throws with perhaps the oddest assertion I've ever seen.